### ISSUE TITLE
Undocumented groups parameter for okta_app_saml

### ISSUE THREAD

**Issue by leonsp-ai** [okta/terraform-provider-okta#167]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.29
+ provider.okta v3.0.38

### Affected Resource(s)

* okta_app_saml 

### Terraform Configuration Files

```hcl
resource "okta_app_saml" "this" {
  groups = ["foo"]
}
```

### Debug Output

N/A

### Panic Output

N/A

### Expected Behavior

The `groups` parameter needs to be documented on https://registry.terraform.io/providers/oktadeveloper/okta/latest/docs/resources/app_saml

### Actual Behavior

The `groups` parameter is not documented

### Steps to Reproduce

1. Visit https://registry.terraform.io/providers/oktadeveloper/okta/latest/docs/resources/app_saml
2. Look for `groups`

### Important Factoids

N/A

### References

* It works the same as the `groups` parameter for `okta_app_oauth`

**Comment by RickB-2840** [comment]
The groups parameter needs more than documentation - I deployed the SAML app and app entitlements for two groups. This populated the groups parameter of the SAML app with the group ids. Users who were members of the groups had app entitlements via the groups.
Bug 1: When I repeated the terraform plan and apply steps, terraform removed the group ids from the SAML app and although the group entitlements still existed, the users in those groups were no longer authorized to use the app. After a Terraform destroy and a new plan and apply, the group id's were again on the SAML application and the users again had access to the app. The bug is that on rerunning Terraform plan, it should not plan to remove the group IDs from the SAML 2.0 application.
Bug 2: If I directly populate the group IDs when I deploy the SAML 2.0 app, then the members of the groups are authorized to the app, but group entitlement objects are not created. 
Bug 3: If I create the SAML 2.0 app with groups on it AND create group entitlement objects, the Okta login page goes offline. https://***********.okta.com/login/login.htm returns error 404 - not found. To recover from this run Terraform destroy.
Given all this weirdness, and the risk of breaking the login page, I recommend the provider keep the app's group ids and the group app entitlements in sync - and if the entitlements still exist, then a Terraform plan should not propose to wipe out the Apps' groups list on reapply.
